RNA-seq/02-gastric_cancer_tximeta.Rmd

```diff
@@ -0,0 +1,213 @@
+
+In this notebook, we'll import the transcript expression quantification output from `salmon quant` using the [`tximeta`](https://bioconductor.org/packages/release/bioc/html/tximeta.html) package.
+`tximeta` is in part a wrapper around another package, [`tximport`](https://bioconductor.org/packages/release/bioc/html/tximport.html), which imports transcript expression data and summarizes it to the gene level.
+Working at the gene rather than transcript level has a number of potential advantages for interpretability, efficiency, and reduction of false positives ([Soneson _et al._ 2016](https://doi.org/10.12688/f1000research.7563.2)).
+`tximeta` eases some of the burden of import by automatically identifying the correct set of annotation data to append to many data sets ([Love _et al._ 2020](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1007664)).
+
+![](diagrams/rna-seq_5.png)
+
+For more information about `tximeta`, see [this excellent vignette](https://www.bioconductor.org/packages/devel/bioc/vignettes/tximeta/inst/doc/tximeta.html) from Love _et al_.
+
+## Libraries and functions
+
+```{r library}
+# Load magrittr for the pipe
+library(magrittr)
+
+# Load the tximeta package
+library(tximeta)
+
+# Load the SummarizedExperiment package
+library(SummarizedExperiment)
+```
+
+## Directories and files
+
+```{r directories, live = TRUE}
+# directory where the data are located
+data_dir <- file.path("data", "gastric-cancer")
+
+# directory where the quant files are located, each sample is its own
+# directory
+quant_dir <- file.path(data_dir, "salmon_quant")
+
+# create a directory to hold the tximport results if it doesn't exist yet
+txi_dir <- file.path(data_dir, "tximport")
+if (!dir.exists(txi_dir)) {
+  dir.create(txi_dir, recursive = TRUE)
+}
+```
+
+We'll need the `quant.sf` files for all the samples in an experiment which we have stored in `quant_dir`. 
+
+```{r input-names}
+# the quant files themselves
+sf_files <- list.files(quant_dir, recursive = TRUE, full.names = TRUE,
+                       pattern = "quant.sf")
+```
+
+```{r metadata-file}
+# sample metadata file
+meta_file <- file.path(data_dir, "gastric-cancer_metadata.tsv")
+```
+
+**Output**
+
+```{r output-names, live = TRUE}
+# Name the output gastric-cancer_tximport.RDS and use the directory created 
+# above as the rest of the path
+txi_out_file <- file.path(txi_dir, "gastric-cancer_tximeta.RDS")
+```
+
+## File names
+
+All output files from `salmon quant` we'll use with `tximeta` are named `quant.sf`.
+Unfortunately, this means that the file names themselves do not have any information about the sample they come from!
+
+```{r sf_files, live = TRUE}
+# Let's look at the full path for the quant.sf files
+sf_files
+```
+
+Let's extract the _sample_ names from the **file paths** using the `stringr` package.
+
+Notice how the file path is separated by `/`.
+If we were to split up this character string by `/`, the second to last item is the sample name.
```

Comment:
I thought we could point out that the sample name here didn't just happen, we chose to use it for the directory ame
```suggestion
If we were to split up this character string by `/`, the second to last item is the sample names (because we used them as  directory names for the `salmon` output).
```